src/Magento/FunctionalTestingFramework/Util/ComposerModuleResolver.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\FunctionalTestingFramework\Util;
+
+use Magento\FunctionalTestingFramework\Composer\Handlers\ComposerInstaller;
+use Magento\FunctionalTestingFramework\Composer\Util\ComposerJsonFinder;
+use Magento\FunctionalTestingFramework\Composer\Objects\ComposerFactory;
+use Magento\FunctionalTestingFramework\Composer\Handlers\ComposerPackager;
+use Magento\FunctionalTestingFramework\Exceptions\TestFrameworkException;
+
+/**
+ * Composer Based Module Resolver
+ */
+class ComposerModuleResolver
+{
+    /**
+     * Code path array from composer json search
+     *
+     * @var array
+     */
+    private $searchedTestModules = null;
+
+    /**
+     * Code path array from composer installed test packages
+     *
+     * @var array
+     */
+    private $installedTestModules = null;
+
+    /**
+     * ComposerModuleResolver constructor
+     */
+    public function __construct()
+    {
+    }
+
+    /**
+     * Get code paths for installed test modules
+     *
+     * @param string $rootComposerFile
+     * @return array
+     * @throws TestFrameworkException
+     */
+    public function getComposerInstalledTestModules($rootComposerFile)
+    {
+        if (null !== $this->installedTestModules) {
+            return $this->installedTestModules;
+        }
+
+        if (!is_file($rootComposerFile) || substr($rootComposerFile, -13) != 'composer.json') {
```

Comment:
use `basename($rootComposerFile)` instead of `substr`

Reply:
Changed